src/adjoint_sensitivity.jl

```diff
@@ -45,17 +45,28 @@ end
 
 # u = λ'
 function (S::ODEAdjointSensitvityFunction)(du,u,p,t)
+  idx = length(S.y)
   y = S.y
-  S.sol(y,t)
-  if isquad(S.alg)
-    λ     = u
-    dλ    = du
-  else
-    idx   = length(y)
+  if isbcksol(S.alg)
```

Comment:
Maybe clean this up through dispatch? Or is there a lot of shared code?

Reply:
Yeah, and even if I use dispatch, I still need to check `isbcksol` and `isquad` to assign variables.